### PR TITLE
[W5.11][W15-A3] Yash Chowdhary

### PR DESCRIPTION
**Pull request by yash-chowdhary**
Added Sort Feature.
'sort' command sorts the persons in the address book in alphabetical order.
Updated Documentation, Help, I/O Tests and JUnit Tests.
Adapted JUnit testing code from #261.

### CODE REVIEW

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -1,9 +1,6 @@
 package seedu.addressbook.data;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (emer7) [inline src/seedu/addressbook/data/AddressBook.java]: It is better to explicitly declare which classes you need to import

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -129,6 +124,23 @@ public void clear() {
...
 
+    /**
+     * Sorts all persons in address book by name
+     * @return success/failure of operation
+     */
+    public boolean sort() {
+        if(internalList.isEmpty()){
+            return false;
+        }
+        Collections.sort(internalList, new Comparator<Person>() {
```
Comment (emer7) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Maybe you can declare a PersonComparator class for a more organized and abstracted code?

File: src/seedu/addressbook/common/PersonComparator.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Comparator class exclusively for comparing persons
+ * in the address book by their names
+ */
+public class PersonComparator implements Comparator<Person>{
+    public PersonComparator(){
+    }
+
+    @Override
+    public int compare(Person o1, Person o2) {
```
Comment (nhatquang3112) [inline src/seedu/addressbook/common/PersonComparator.java]: can change the naming from 'o1' and 'o2' to 'firstPerson' and 'secondPerson' for better readability

Comment (nhatquang3112) [inline src/seedu/addressbook/common/PersonComparator.java:16]: can change the naming from 'o1' and 'o2' to 'firstPerson' and 'secondPerson' for better readability

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ */
+public class SortCommand extends Command{
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts all persons in the address book in alphabetical order of their name.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "All persons in address book sorted by name";
+    public static final String MESSAGE_FAILURE = "Unable to sort persons in address book.";
```
Comment (ktingit) [inline src/seedu/addressbook/commands/SortCommand.java:14]: Good job on a whole! Perhaps you could consider changing this line to something like "There are no persons in address book to sort." to be more specific and more user-friendly?

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -129,6 +130,18 @@ public void clear() {
         internalList.clear();
     }
 
+    /**
+     * Sorts all persons in address book by name
+     * @return success/failure of operation
+     */
+    public boolean sort() {
+        if(internalList.isEmpty()){
```
Comment (eldriclim) [inline src/seedu/addressbook/data/person/UniquePersonList.java:138]: Such statement requires a whitespace after keywords like `if` `while`.
You may leave it as it is for now, however in 4kloc, such coding standards will be enforced, you might want to refer to the coding standards for the module.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package seedu.addressbook.commands;
+
+import org.junit.Test;
```
Comment (eldriclim) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:3]: Good job on JUnit test

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    public void sortCommand_addressBookWithFiveElements_sortedAddressBook(){
+        SortCommand sortCommand = new SortCommand();
+
+        Person miguel = TestUtil.generateTestPerson("Miguel");
+        Person suzanne = TestUtil.generateTestPerson("Suzanne");
+        Person carley = TestUtil.generateTestPerson("Carley");
+        Person genna = TestUtil.generateTestPerson("Genna");
+        Person huey = TestUtil.generateTestPerson("Huey");
+
+        AddressBook addressBook = TestUtil.createAddressBook(miguel,suzanne,carley,genna,huey);
```
Comment (eldriclim) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:27]: Likewise for `parameters`, remember to follow coding standards

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        List<Person> sortedList = Arrays.asList(carley,genna,huey,miguel,suzanne);
+
+        sortCommand.setData(addressBook, Collections.emptyList());
+        sortCommand.execute();
+
+        UniquePersonList personList = addressBook.getAllPersons();
+        UniquePersonList expectedList = null;
+        try{
```
Comment (eldriclim) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:35]: It is better to have a separate JUnit test solely for testing exceptions

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+            expectedList = new UniquePersonList(sortedList);
+        } catch (Exception e) {
```
Comment (eldriclim) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:37]: Avoid catching exceptions by `Exception`

**Review by eldriclim — COMMENTED** [review]
Good job on clearing 2kloc requirements. Please close PR after reading the comments.

**Pull request closed without merging**